File: backend/src/main/kotlin/com/budgetbook/feedback/domain/ReleaseNote.kt
package com.budgetbook.feedback.domain

import com.budgetbook.auth.domain.User
import com.budgetbook.common.entity.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.Instant
import java.util.UUID

@Entity
@Table(name = "release_notes")
class ReleaseNote(
    @Id
    val id: UUID = UUID.randomUUID(),

    @Column(nullable = false, length = 20)
    var version: String,

    @Column(nullable = false, length = 200)
    var title: String,

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String,

    @Column(name = "is_published", nullable = false)
    var isPublished: Boolean = false,

    @Column(name = "published_at")
    var publishedAt: Instant? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    val createdBy: User,

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "release_note_feedbacks",
        joinColumns = [JoinColumn(name = "release_note_id")],
        inverseJoinColumns = [JoinColumn(name = "feedback_post_id")]
    )
    val linkedFeedbacks: MutableSet<FeedbackPost> = mutableSetOf()
) : BaseTimeEntity()

File: frontend/lib/features/feedback/domain/entities/release_note.dart
import 'package:equatable/equatable.dart';

class ReleaseNote extends Equatable {
  final String id;
  final String version;
  final String title;
  final String content;
  final bool isPublished;
  final DateTime? publishedAt;
  final List<String> linkedFeedbackIds;
  final DateTime createdAt;

  const ReleaseNote({
    required this.id,
    required this.version,
    required this.title,
    required this.content,
    required this.isPublished,
    this.publishedAt,
    required this.linkedFeedbackIds,
    required this.createdAt,
  });

  @override
  List<Object?> get props => [
        id,
        version,
        title,
        content,
        isPublished,
        publishedAt,
        linkedFeedbackIds,
        createdAt,
      ];
}
